### PR TITLE
 No Response on Initial Message problem fixed and sound feature added

### PR DESCRIPTION
**Pull request by amansoni7477030**
1. Fxed a major issue with responses, as defined in https://github.com/FlowiseAI/Flowise/issues/2291 and in issue #158. Now, it will always show a response, even if the server is very slow with lower CPU and RAM. If the websocket takes time to connect and streaming is already over, it will directly show the result.
2. Added a new feature of sound: now users can add input and output sound effects for sending and receiving messages. Below are the parameters used:
```
sendMessageSound: true,
// sendSoundLocation: "send_message.mp3", // If not used, the default sound effect will play when sendMessageSound is true.
receiveMessageSound: true,
// receiveSoundLocation: "receive_message.mp3", // If not used, the default sound effect will play when receiveMessageSound is true.
```

### CODE REVIEW

**Comment by HenryHengZJ** [comment]
Now we store the base64 string of the sound, it makes the `web.js` becomes extremely long: 
image

perhaps we could store it to like `raw.github` or some other storage, instead of local

File: src/components/Bot.tsx [hunk 1]
```diff
@@ -429,9 +495,9 @@ export const Bot = (botProps: BotProps & { class?: string }) => {
         else if (data.json) text = JSON.stringify(data.json, null, 2);
         else text = JSON.stringify(data, null, 2);
 
-        updateLastMessage(text, data?.chatMessageId, data?.sourceDocuments, data?.fileAnnotations, data?.agentReasoning);
+        updateLastMessage(text, data?.chatMessageId, data?.sourceDocuments, data?.fileAnnotations, data?.agentReasoning, data.text);
       } else {
-        updateLastMessage('', data?.chatMessageId, data?.sourceDocuments, data?.fileAnnotations, data?.agentReasoning);
+        updateLastMessage('', data?.chatMessageId, data?.sourceDocuments, data?.fileAnnotations, data?.agentReasoning, data.text);
```
Comment (HenryHengZJ) [inline src/components/Bot.tsx:500]: when not streaming, do we have `data.text`?

**Comment by amansoni7477030** [comment]
> Now we store the base64 string of the sound, it makes the `web.js` becomes extremely long: image
> 
> perhaps we could store it to like `raw.github` or some other storage, instead of local

@HenryHengZJ Initially, I used GitHub Raw, which worked well. However, I started receiving a warning about "Third-party cookies being blocked in future Chrome versions as part of Privacy Sandbox." This means that after 2024, it may not work directly in Chrome, and we'll need to use some workarounds. Therefore, I reconsidered and decided that using CDN is the best option, although it can sometimes be slow. That's why I switched to using local storage. As you suggested, I've updated the code so that it now defaults to using CDN. If needed, you can modify the CDN link.

File: src/components/Bot.tsx (hunk 1 above)
Comment (amansoni7477030) [reply src/components/Bot.tsx:485]: yes

Comment (amansoni7477030) [reply src/components/Bot.tsx:485]: @HenryHengZJ In the code, there is a rare condition where even if the if (i === data.length - 1) part executes, sometimes there is no output or the UI remains blank. To address this issue, I added a condition to check if resultText is available but the UI hasn't updated (!uiUpdated). This ensures that the UI always displays the output. Please check the attached image where the first condition was executed but did not produce any output, hence the second condition was also executed.

**Review by HenryHengZJ — APPROVED** [review]

**Pull request merged**